core/src/main/java/io/undertow/server/handlers/resource/DirectoryUtils.java

```diff
@@ -154,33 +183,89 @@ public static StringBuilder renderDirectoryListing(final HttpServerExchange exch
                 parent = "/";
             }
         }
+        return parent;
+    }
+
+    private static void appendFiles(List<Resource> files, StringBuilder builder, int i, String sortUrl, SimpleDateFormat format) {
+        for (Resource entry : files) {
+            builder.append("<tr class='").append((++i & 1) == 1 ? "odd" : "even").append("'><td><a class='icon file' href='")
+                    .append(sortUrl).append(entry.getName()).append("'>")
+                    .append(entry.getName()).append("</a></td><td>")
+                    .append(format.format((entry.getLastModified() == null) ? new Date(0L) : entry.getLastModified()))
+                    .append("</td><td>");
+            formatSize(builder, entry.getContentLength());
+            builder.append("</td></tr>\n");
+        }
+    }
 
-        SimpleDateFormat format = new SimpleDateFormat("MMM dd, yyyy HH:mm:ss", Locale.US);
-        int i = 0;
-        if (parent != null) {
-            i++;
-            builder.append("<tr class='odd'><td><a class='icon up' href='").append(relative  == null ? parent : relative + parent).append(parent.endsWith("/") ? "" : "/").append("'>[..]</a></td><td>");
-            builder.append(format.format((resource.getLastModified() == null ? new Date(0L) : resource.getLastModified())))
+    private static void appendDirectories(List<Resource> directories, StringBuilder builder, int i, String sortUrl, SimpleDateFormat format) {
+        for (Resource entry : directories) {
+            builder.append("<tr class='").append((++i & 1) == 1 ? "odd" : "even").append("'><td><a class='icon dir' href='")
+                    .append(sortUrl).append(entry.getName()).append("/'>")
+                    .append(entry.getName()).append("</a></td><td>")
+                    .append(format.format((entry.getLastModified() == null) ? new Date(0L) : entry.getLastModified()))
                     .append("</td><td>--</td></tr>\n");
         }
+    }
+
+    private static Comparator<Resource> getComparator(String sortColumn, String currentSortOrder) {
+        Comparator<Resource> comparator;
+        if ("lastModified".equals(sortColumn)) {
+            comparator = Comparator.comparing(
+                    entry -> (entry.getLastModified() == null) ? new Date(0L) : entry.getLastModified()
```

Comment:
How this can be null?

Reply:
if its somehow possible, maybe just use creation date rather than 1970 ?

Reply:
There is no creation date, but I think using a string like `Unknown` or `Not Available`. What do you think?

Reply:
It makes sense to me. As we sure that it is possible to get a null here? Maybe we even want to have something like: "-"
Such as: last modified: -